src/protocols/bdx/BDXMessageUtils.cpp

```diff
@@ -185,22 +158,35 @@ CHIP_ERROR TransferInit::Parse(System::PacketBuffer * aBuffer, TransferInit & aP
 
     SuccessOrExit(bufReader.Read16(&aParsedMessage.mFileDesLength).StatusCode());
 
-    VerifyOrExit(bufReader.OctetsRead() + aParsedMessage.mFileDesLength <= aBuffer->DataLength(),
+    VerifyOrExit(bufReader.OctetsRead() + aParsedMessage.mFileDesLength <= aBuffer.DataLength(),
                  err = CHIP_ERROR_MESSAGE_INCOMPLETE);
     aParsedMessage.mFileDesignator = &bufStart[bufReader.OctetsRead()];
 
     // Rest of message is metadata (could be empty)
-    if (bufReader.OctetsRead() + aParsedMessage.mFileDesLength < aBuffer->DataLength())
+    if (bufReader.OctetsRead() + aParsedMessage.mFileDesLength < aBuffer.DataLength())
     {
-        uint16_t metadataStartIndex    = (uint16_t)(bufReader.OctetsRead() + aParsedMessage.mFileDesLength);
+        uint16_t metadataStartIndex    = static_cast<uint16_t>(bufReader.OctetsRead() + aParsedMessage.mFileDesLength);
         aParsedMessage.mMetadata       = &bufStart[metadataStartIndex];
-        aParsedMessage.mMetadataLength = (uint16_t)(aBuffer->DataLength() - metadataStartIndex);
+        aParsedMessage.mMetadataLength = static_cast<uint16_t>(aBuffer.DataLength() - metadataStartIndex);
     }
 
 exit:
+    if (bufReader.StatusCode() != CHIP_NO_ERROR)
```

Comment:
I feel like the overall "SuccessOrExit/VerifyOrExit" has very little added value in these functions where no resources are to be released at the end. Early return would, in my opinion, be much simpler, and would allow you to have scope values such as `tmpUint32Value` within a subblock, which now you have to forcibly put at the top because of the goto crossing of OrExit macros. Overall, it prevents proper scoping of temporary variable usage in a way that would make the code easier to understand.